test/integration/scheduler/util.go

```diff
@@ -61,49 +63,80 @@ import (
 )
 
 type testContext struct {
-	closeFn             framework.CloseFunc
-	httpServer          *httptest.Server
-	ns                  *v1.Namespace
-	clientSet           *clientset.Clientset
-	informerFactory     informers.SharedInformerFactory
-	schedulerConfigArgs *factory.ConfigFactoryArgs
-	schedulerConfig     *factory.Config
-	scheduler           *scheduler.Scheduler
-	stopCh              chan struct{}
-}
-
-// createConfiguratorWithPodInformer creates a configurator for scheduler.
-func createConfiguratorArgsWithPodInformer(
+	closeFn         framework.CloseFunc
+	httpServer      *httptest.Server
+	ns              *v1.Namespace
+	clientSet       *clientset.Clientset
+	informerFactory informers.SharedInformerFactory
+	scheduler       *scheduler.Scheduler
+	stopCh          chan struct{}
+}
+
+// createSchedulerWithPodInformer creates a new scheduler.
```

Comment:
this method only has 2 calls, and they set different parameters... maybe we should just call `scheduler.New` directly in each of them.

Reply:
Having a method that basically takes most of the schefuler fields and creates a scheduler to serve different cases of creating the scheduler makes sense for me and could later be used by other tests. What do you think? (I also get the point of removing the method)

Reply:
Note that one of the callers of the functions is already reusable `initTestSchedulerWithOptions`. Maybe the test that is using `createSchedulerWithPodInformer` directly should be using that instead. But if it's special enough, I'm happy to have it call `scheduler.New` directly.

Reply:
sg, done! 
The only reason not to use `initTestScheduler` is the scheduler name (fooScheduler). Not sure if this could be changed in order to use `initTestScheduler`

Reply:
Interestingly enough, `initTestSchedulerWithOptions` doesn't receive any `scheduler.Options`. That can be the way to pass the name. And also `WithDisabledPreemption` can be used to remove that parameter.

Reply:
Do you think this should be changed in this PR?

Reply:
Yes, it's highly related and small enough.